savageRoller.ino

```diff
@@ -668,46 +672,59 @@ void splashHandleKeys() {
       displayDeck();
     } else if (isNewlyPressed('b')) {
       currentPage = Page::Battery;
-      firstRun = 1;
+      last_battery_check = 0;
+      batteryHandleDisplay();
     }
   }
 }
 
 /* Display the currently estimated battery percentage on the screen. */
 void batteryHandleDisplay() {
-  if (!stateChange) {
+  // Only update the display if it's been more then 5000 clocks since last check
+  std::clock_t now = std::clock();
+  if (now - last_battery_check < 5000) {
```

Comment:
The `now` value is in clock ticks per the [library function definition](https://en.cppreference.com/w/cpp/chrono/c/clock), and not necessarily milliseconds. You need to divide this by [CLOCKS_PER_SEC](https://en.cppreference.com/w/cpp/chrono/c/CLOCKS_PER_SEC) to get seconds.